src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -221,6 +235,30 @@ public AgentSettings LoadSettings()
                 }
             }
 
+            // See if the server supports our OAuth key exchange for credentials
+            if (agent.Authorization != null && 
+                agent.Authorization.ClientId != Guid.Empty &&
+                agent.Authorization.AuthorizationUrl != null)
+            {
+                var credentialData = new CredentialData
+                {
+                    Scheme = Constants.Configuration.OAuth,
+                    Data =
+                    {
+                        { "clientId", agent.Authorization.ClientId.ToString("D") },
```

Comment:
No, in WebApi they are first-class properties. Since I think we still want to support talking to servers that do not support OAuth, it made sense to me to simply plug into the existing credential architecture.